### PR TITLE
[CI] Improve workflows

### PR DESCRIPTION
**Pull request by TomasVotruba**
Last follow up to https://github.com/sabbelasichon/typo3-rector/pull/1962

Tip for automated weekly coverage: https://github.com/rectorphp/rector/blob/master/.github/workflows/weekly_code_coverage.yaml

<br>

Btw, what do you use coverage in practise? I'm thinking about it lately and I never work with the coverage number in any way :)

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -11,25 +11,20 @@ jobs:
...
-            -   uses: actions/cache@v1
-                with:
-                    path: ${{ steps.composer-cache.outputs.dir }}
-                    key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
-                    restore-keys: |
-                        ${{ runner.os }}-composer-
+
             -   uses: shivammathur/setup-php@v2
                 with:
-                    php-version: 7.3
```
Comment (TomasVotruba) [inline .github/workflows/tests.yaml:29]: It seems tests only run on PHP 7.3.
That explain why I had so many errors on PHP 8.0 :)

File: .github/workflows/code_coverage.yaml [hunk 2]
```diff
@@ -11,22 +10,16 @@ jobs:
...
             -   uses: shivammathur/setup-php@v2
                 with:
                     php-version: 7.3
                     coverage: none
                 env:
                     COMPOSER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-            -   run: composer update --no-progress
+            # see https://github.com/ramsey/composer-install
+            -   uses: "ramsey/composer-install@v1"
```
Comment (TomasVotruba) [inline .github/workflows/code_coverage.yaml:22]: Little cool trick to simplify cached composer install

**Comment by TomasVotruba** [comment]
Ready to review :heavy_check_mark:

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (sabbelasichon) [reply .github/workflows/tests.yaml:29]: @TomasVotruba Oh no, stupid me. Thanks for fixing.

**Pull request merged**